spec/delete_host_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rails_helper' # spec_helper?
+require 'sidekiq/testing'
+
+describe DeleteHost do
+  let(:user) { FactoryBot.create(:user) }
+  let(:host) { FactoryBot.create(:host, org_id: user.account.org_id) }
+  let(:profile) { FactoryBot.create(:profile, :with_rules, account: user.account) }
+  let(:message) { {'id' => host.id, 'type' => 'delete'} }
+
+  before do
+    logger = double(Rails.logger)
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:debug)
+    allow(Sidekiq).to receive(:logger)
+  end
+
+  DeleteHost.clear
+
+  context 'when the deletion succeeds' do
+    let(:test_result) { FactoryBot.create(:test_result, profile: profile, host: host) }
+    let(:rule_result) { FactoryBot.create(:rule_result, rule: profile.rules.sample, test_result: test_result, host: host) } 
+
+    it 'finds a host to delete' do
+      expect(profile.policy.reload.test_result_host_count).not_to equal(0)
+      expect(profile.reload.score).not_to equal(0)
+    end
+
+    it 'logs and enqueues deletion' do # TODO Deleteted????
+      expect(Rails.logger).to receive(:audit_success).with("Deleteted related records for host #{host_id}")
+      
+      DeleteHost.perform_async(message)
+      
+      expect(DeleteHost.jobs.size).to equal(1)  
+    end
+
+    it 'performs deletion' do
+      expect { 
+        DeleteHost.drain 
+      }.to change { test_result }.by(-1)
+       .and change { rule_result }.by(-1)
```

Comment:
This is an interesting syntax, I would probably not go with it. Simplify:
```suggestion
      expect(test_result.count).to eq(n)
      expect(rule_result.count).to eq(n)
      DeleteHost.drain
      expect(test_result.count).to eq(n-1)
      expect(rule_result.count).to eq(n-1)
```

But IMO we can do much better tests and actually query the DB that ALL the test and rule results for the hosts are nonexistent afterwards.